inject/inject.vala

```diff
@@ -0,0 +1,415 @@
+namespace Frida.Inject {
+	private static Application application;
+
+	private static bool output_version;
+	private static bool disable_jit;
+	private static bool enable_development;
+	private static int pid;
+	private static string script_file;
+
+	static const OptionEntry[] options = {
+		{ "version", 0, 0, OptionArg.NONE, ref output_version, "Output version information and exit", null },
+		{ "disable-jit", 0, 0, OptionArg.NONE, ref disable_jit, "Disable the JIT runtime", null },
+		{ "development", 'D', 0, OptionArg.NONE, ref enable_development, "Enable development mode", null },
+		{ "pid", 'p', 0, OptionArg.INT, ref pid, null, "PID" },
+		{ "script", 's', 0, OptionArg.FILENAME, ref script_file, null, "JAVASCRIPT_FILENAME" },
+		{ null }
+	};
+
+	private static int main (string[] args) {
+#if !WINDOWS
+		Posix.setsid ();
+#endif
+
+		Environment.init ();
+
+		pid = -1;
+		script_file = "";
+
+		try {
+			var ctx = new OptionContext ();
+			ctx.set_help_enabled (true);
+			ctx.add_main_entries (options, null);
+			ctx.parse (ref args);
+
+			if (output_version) {
+				stdout.printf ("%s\n", version_string ());
+				return 0;
+			}
+		} catch (OptionError e) {
+			stdout.printf ("%s\n", e.message);
+			stdout.printf ("Run '%s --help' to see a full list of available command line options.\n", args[0]);
+			return 1;
+		}
+
+		if (pid == -1) {
+			stdout.printf ("pid must be specified\n");
+			return 1;
+		}
+
+		if (script_file == "") {
+			stdout.printf ("javascript filename must be specified\n");
```

Comment:
All of these errors should be using `printerr`, and on line 36 we can use `print`.